### PR TITLE
feat: add spark 3.2.1

### PR DESCRIPTION
**Pull request by guangie88**
Drop Spark 3.0.z to keep only the last 3 version (including 2.4.8).

Update 3.1.z to 3.1.2.

### CODE REVIEW

File: templates/vars.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  scala:  ['2.11', '2.12']
-
-- spark:  ['3.0.3', '3.1.2']
+- spark:  ['3.0.3', '3.1.3']
   java:   ['8', '11']
   hadoop: ['2.7.4', '3.2.0']
   scala:  ['2.12']
+
+- spark:  ['3.2.1']
+  java:   ['8', '11']
```
Comment (xtrntr) [inline templates/vars.yml:10]: Is it possible to confirm what version of java 8 would be used in the image?

> Java 8 prior to version 8u201 support is deprecated as of Spark 3.2.0.
You will need to use a compatible Scala version (2.12.x).

https://spark.apache.org/docs/latest/

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -123,7 +123,7 @@ jobs:
     - name: Install setuptools and wheel
       run: python3 -m pip install setuptools wheel
     - name: Build Docker image
-      run: bash make-distribution.sh
+      run: MAVEN_OPTS="-Xss2M" bash make-distribution.sh
```
Comment (guangie88) [inline .github/workflows/ci.yml:126]: Keep encountering StackOverflow exception, so followed this:
https://www.lagomframework.com/documentation/1.6.x/java/JVMMemoryOnDev.html

Log:

```log
2022-04-04T03:17:31.1979950Z Exception in thread "main" java.lang.StackOverflowError
2022-04-04T03:17:31.1980211Z 	at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2504)
2022-04-04T03:17:31.1980475Z 	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$103(Typers.scala:5711)
2022-04-04T03:17:31.1980773Z 	at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:500)
2022-04-04T03:17:31.1981020Z 	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5746)
2022-04-04T03:17:31.1981259Z 	at scala.tools.nsc.transform.Erasure$Eraser.typed1(Erasure.scala:809)
2022-04-04T03:17:31.1981502Z 	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5781)
2022-04-04T03:17:31.1981756Z 	at scala.tools.nsc.transform.Erasure$Eraser.adaptMember(Erasure.scala:734)
2022-04-04T03:17:31.1981991Z 	at scala.tools.nsc.transform.Erasure$Eraser.typed1(Erasure.scala:809)
2022-04-04T03:17:31.1982233Z 	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5781)
2022-04-04T03:17:31.1982490Z 	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$39(Typers.scala:4891)
2022-04-04T03:17:31.1982723Z 	at scala.tools.nsc.typechecker.Typers$Typer.silent(Typers.scala:713)
2022-04-04T03:17:31.1982986Z 	at scala.tools.nsc.typechecker.Typers$Typer.normalTypedApply$1(Typers.scala:4893)
2022-04-04T03:17:31.1983338Z 	at scala.tools.nsc.typechecker.Typers$Typer.typedApply$1(Typers.scala:4921)
2022-04-04T03:17:31.1983575Z 	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5736)
2022-04-04T03:17:31.1983801Z 	at scala.tools.nsc.transform.Erasure$Eraser.typed1(Erasure.scala:809)
2022-04-04T03:17:31.1984025Z 	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5781)
2022-04-04T03:17:31.1984277Z 	at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2498)
2022-04-04T03:17:31.1984539Z 	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$103(Typers.scala:5711)
2022-04-04T03:17:31.1984831Z 	at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:500)
...
```

File: templates/vars.yml (hunk 1 above)
Comment (guangie88) [reply templates/vars.yml:10]: Yes, based on the GHA:

```
 Java configuration:
  Distribution: adopt
  Version: 8.0.292+1
  Path: /opt/hostedtoolcache/Java_Adopt_jdk/8.0.292-1/x64
```

Ref: https://github.com/dsaidgovsg/spark-k8s/runs/6127369063?check_suite_focus=true

**Comment by guangie88** [comment]
Will go ahead to merge and build the images, since the Java 8 is confirmed to be within the supported range:
https://github.com/dsaidgovsg/spark-k8s/pull/35#discussion_r856221304

**Pull request merged**